internal/controller/managedcluster_controller.go

```diff
@@ -66,6 +67,12 @@ type ManagedClusterReconciler struct {
 	SystemNamespace string
 }
 
+type objectToCheck struct {
```

Comment:
I think this type doesn't required to be on the pkg level, and also may be aligned
```go
type objectToCheck struct {
	statuses           map[string]string
	gvr                schema.GroupVersionResource
	requeueIfNoObjects bool
}
```

Reply:
We still need "statuses" field to filter by only the statuses we wanted to check against.

Reply:
This type isn't required to be package wide. Besides, two strings could be checked in a single if statement as I have already suggested earlier, so I don't see the need in this type in general. Wdyt?

Reply:
My bad: the type is indeed required to be defined as a global. Either way consider to stick to another approach. It's up to you though